File: app/src/main/java/com/scatl/uestcbbs/module/message/adapter/SystemMsgAdapter.java
package com.scatl.uestcbbs.module.message.adapter;

import android.view.View;
import android.widget.Button;

import com.chad.library.adapter.base.BaseQuickAdapter;
import com.chad.library.adapter.base.BaseViewHolder;
import com.scatl.uestcbbs.R;
import com.scatl.uestcbbs.entity.SystemMsgBean;
import com.scatl.uestcbbs.helper.glidehelper.GlideLoader4Common;
import com.scatl.uestcbbs.util.TimeUtil;

/**
 * author: sca_tl
 * description:
 * date: 2020/1/27 15:55
 */
public class SystemMsgAdapter extends BaseQuickAdapter<SystemMsgBean.BodyBean.DataBean, BaseViewHolder> {
    public SystemMsgAdapter(int layoutResId) {
        super(layoutResId);
    }

    @Override
    protected void convert(BaseViewHolder helper, SystemMsgBean.BodyBean.DataBean item) {
        helper.setText(R.id.user_name, item.user_name)
                .setText(R.id.content, item.note)
                .setText(R.id.date, TimeUtil.formatTime(item.replied_date, R.string.post_time1, mContext))
                .addOnClickListener(R.id.user_icon)
                .addOnClickListener(R.id.action_btn);
        GlideLoader4Common.simpleLoad(mContext, item.icon, helper.getView(R.id.user_icon));

        Button button = helper.getView(R.id.action_btn);
        if (item.actions != null && item.actions.size() != 0) {
            button.setVisibility(View.VISIBLE);
            button.setText(item.actions.get(0).title);
        } else {
            button.setVisibility(View.GONE);
        }
//        if (item.has_action) {
//            helper.getView(R.id.item_system_action_btn).setVisibility(View.VISIBLE);
//            helper.setText(R.id.item_system_action_btn, item.actionBean.get(0).title);
//        } else {
//            helper.getView(R.id.item_system_action_btn).setVisibility(View.GONE);
//        }
    }
}

File: app/src/main/java/com/scatl/uestcbbs/module/message/adapter/SystemMsgAdapter.kt
package com.scatl.uestcbbs.module.message.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.scatl.uestcbbs.databinding.ItemSystemMsgBinding
import com.scatl.uestcbbs.entity.SystemMsgBean
import com.scatl.uestcbbs.helper.PreloadAdapter
import com.scatl.uestcbbs.helper.ViewBindingHolder
import com.scatl.uestcbbs.util.TimeUtil
import com.scatl.uestcbbs.R
import com.scatl.uestcbbs.util.isNullOrEmpty
import com.scatl.uestcbbs.util.load

/**
 * Created by sca_tl at 2023/6/12 15:01
 */
class SystemMsgAdapter: PreloadAdapter<SystemMsgBean.BodyBean.DataBean, ItemSystemMsgBinding>() {

    override fun getViewBinding(parent: ViewGroup): ItemSystemMsgBinding {
        return ItemSystemMsgBinding.inflate(LayoutInflater.from(context), parent, false)
    }

    override fun onBindViewHolder(holder: ViewBindingHolder<ItemSystemMsgBinding>, position: Int, item: SystemMsgBean.BodyBean.DataBean?) {
        super.onBindViewHolder(holder, position, item)
        if (item == null) {
            return
        }

        holder.binding.userName.text = item.user_name
        holder.binding.content.text = item.note
        holder.binding.date.text = TimeUtil.formatTime(item.replied_date, R.string.post_time1, context)
        holder.binding.userIcon.load(item.icon)

        if (item.actions.isNullOrEmpty()) {
            holder.binding.actionBtn.visibility = View.GONE
        } else {
            holder.binding.actionBtn.visibility = View.VISIBLE
            holder.binding.actionBtn.text = item.actions[0].title
        }
    }
}